distage/distage-framework-docker/src/main/scala/izumi/distage/docker/FileLockMutex.scala

```diff
@@ -47,10 +42,23 @@ object FileLockMutex {
         val file = new File(s"$tmpDir/$filename.tmp")
         val newFileCreated = file.createNewFile()
         if (newFileCreated) file.deleteOnExit()
-        FileChannel.open(file.toPath, StandardOpenOption.WRITE)
+        AsynchronousFileChannel.open(file.toPath, StandardOpenOption.WRITE)
       }
     )(release = channel => F.definitelyRecover(F.maybeSuspend(channel.close()))(_ => F.unit)) {
-      acquireAndRun(_, 0)
+      channel =>
+        retryOnFileLock {
+          P.async[Option[FileLock]] {
+            cb =>
+              val handler = new CompletionHandler[FileLock, Unit] {
+                override def completed(result: FileLock, attachment: Unit): Unit = cb(Right(Some(result)))
+                override def failed(exc: Throwable, attachment: Unit): Unit = cb(Left(exc))
+              }
+              channel.lock((), handler)
+          }
+        }.flatMap {
```

Comment:
This looks like it should use `bracket` instead

Reply:
Done.